src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/JsonFileDeserializer.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using OmniSharp.Extensions.LanguageServer.Protocol.Serialization;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal abstract class JsonFileDeserializer
+    {
+        public static readonly JsonFileDeserializer Instance = new DefaultJsonFileDeserializer();
+
+        public abstract TValue Deserialize<TValue>(string filePath);
+
+        private class DefaultJsonFileDeserializer : JsonFileDeserializer
+        {
+            public override TValue Deserialize<TValue>(string filePath)
+            {
+                using (var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete))
+                using (var reader = new StreamReader(stream))
+                {
+                    var deserializedValue = (TValue)Serializer.Instance.JsonSerializer.Deserialize(reader, typeof(TValue));
```

Comment:
Should try-catch around this. Looks like failure to deserialize can occasionally throw. We don't want that, we'd rather return `null` here.